docs: add testing strategy (#2589)
Sage/carbon: cypress/README.md

# Testing Strategy

## Unit Testing

> Unit tests are writing / maintenancing by the developers

Reviewer: ```suggestion
Unit tests are written and maintained by developers.
```

Author: updated

## Manual Testing
### Components:
1. Every component should be fully tested using functional testing / UI testing / Internationalisation testing

> 2. Should be tested all stories for dedicated component (default / classic / multiple / validation / themes etc.), notes section and accessibility section.

Reviewer 2: Please, remove the blank space, it'll look better :)

Author: fixed

> 2. Should be tested all stories for dedicated component (default / classic / multiple / validation / themes etc.), notes section and accessibility section.

Reviewer: ```suggestion
2. Each component should have a dedicated story (default / classic / multiple / validation / themes etc.), notes section and accessibility section.
```

Author: updated

3. Supporting browsers:
  * Chrome
  * Firefox
  * Safari
  * Edge
  * Internet Explorer 11
4. Jira:

>   * For every testable task needs to be created test sessions per every browser

Reviewer: ```suggestion
  * For every testable task, a JIRA test session needs to be created for each of the supported browsers.
```

Author: updated

  * Results should be added to the created test sessions with pass / fail mark

>   * If the result is set to fail the issue should be raised in Jira

Reviewer: ```suggestion
  * Any issues found during testing should be raised as a "Defect" JIRA.
  * If the issue is present on the `master` branch it should be raised as a "Bug" in JIRA, not "Defect".
```

Author: updated

# Cypress Cucumber tests for Carbon storybook (Automation Testing)

## Description

>   * Automation testing are writing using Cypress.io framework.

Reviewer: ```suggestion
  * Automation tests are written using [Cypress.io](https://www.cypress.io/) framework.
```

Author: updated

# Testing Strategy

## Unit Testing
   Unit tests are written and maintained by developers.

## Manual Testing
### Components:
1. Every component should be fully tested using functional testing / UI testing / Internationalisation testing.

> 2. Each component should have a dedicated story (default / classic / multiple / validation / themes etc.),

Reviewer: ```suggestion
2. Every component should have a dedicated story (default / classic / multiple / validation / themes etc.),
```

Author: fixed

   notes section and accessibility section.
3. Supporting browsers:
  * Chrome,
  * Firefox,
  * Safari,
  * Edge,
  * Internet Explorer 11.
4. Jira:
  * For every testable task, a JIRA test session needs to be created for each of the supported browsers;
  * Results should be added to the created test sessions with pass / fail mark;
  * Any issues found during testing should be raised as a "Defect" JIRA;
  * If the issue is present on the `master` branch it should be raised as a "Bug" in JIRA, not as a "Defect".

# Cypress Cucumber tests for Carbon storybook (Automation Testing)

## Description

>   * Automation tests are written using [Cypress.io] (https://www.cypress.io/) framework.

Reviewer: ```suggestion
  * Automation tests are written using [Cypress.io](https://www.cypress.io/) framework.
```

Author: updated
